dev/tests/js/jasmine/tests/app/code/Magento/Ui/frontend/js/model/messages.test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'uiRegistry',
+    'Magento_Ui/js/model/messages'
+], function (registry, Constr) {
+    'use strict';
+
+    describe('Magento_Ui/js/model/messages', function () {
+        var obj = new Constr({
+            provider: 'provName',
+            name: '',
+            index: ''
+        });
+
+        registry.set('provName', {
+            /** Stub */
+            on: function () {
+            },
+
+            /** Stub */
+            get: function () {
+            },
+
+            /** Stub */
+            set: function () {
+            }
+        });
+
+        describe('initialize method', function () {
+            it('check for existing', function () {
+                expect(obj).toBeDefined();
+            });
+        });
+
+        describe('add method', function () {
+            it('simple message', function () {
+                var messageObj = {
+                        message: 'Message test'
+                    },
+                    type = [],
+                    returnedObj = ['Message test'];
+
+                expect(obj.add(messageObj, type)).toEqual(true);
+                expect(type).toEqual(returnedObj);
+            });
+
+            it('message with parameters', function () {
+                var messageObj = {
+                        message: 'Message test case %1, case %2 and case %3',
+                        parameters: [
+                            'one',
+                            'two',
+                            'three'
+                        ]
+                    },
+                    type = [],
+                    returnedObj = ['Message test case ' + messageObj.parameters[0] + ', case ' +
+                    messageObj.parameters[1] + ' and case ' + messageObj.parameters[2]];
+
+                expect(obj.add(messageObj, type)).toEqual(true);
+                expect(type).toEqual(returnedObj);
+            });
+        });
+
+        describe('check methods: hasMessages, addErrorMessage, getErrorMessages', function () {
+            var errorMessageText = 'Error message test';
+
+            it('hasMessages method before adding messages', function () {
+                expect(obj.hasMessages()).toEqual(false);
+            });
+
+            it('check addErrorMessage method', function () {
+                var messageObj = {
+                    message: errorMessageText
+                };
+
+                expect(obj.addErrorMessage(messageObj)).toEqual(true);
+            });
+
+            it('check getErrorMessage method', function () {
+                expect(obj.getErrorMessages()()).toEqual([errorMessageText]);
+            });
+
+            it('hasMessages method after adding Error messages', function () {
+                expect(obj.hasMessages()).toEqual(true);
```

Comment:
Every test here should be isolated and not depend on other tests being executed before or after it.